tracer/src/Datadog.Trace/Util/Http/HttpRequestExtensions.Core.cs

```diff
@@ -78,40 +77,17 @@ internal static partial class HttpRequestExtensions
 
             var dict = new Dictionary<string, object>
             {
-                {
-                    AddressesConstants.RequestMethod, request.Method
-                },
-                {
-                    AddressesConstants.RequestUriRaw, url
-                },
-                {
-                    AddressesConstants.RequestQuery, queryStringDic
-                },
-                {
-                    AddressesConstants.RequestHeaderNoCookies, headersDic
-                },
-                {
-                    AddressesConstants.RequestCookies, cookiesDic
-                },
+                { AddressesConstants.RequestMethod, request.Method },
+                { AddressesConstants.RequestUriRaw, request.GetUrl() },
+                { AddressesConstants.RequestQuery, queryStringDic },
+                { AddressesConstants.RequestHeaderNoCookies, headersDic },
+                { AddressesConstants.RequestCookies, cookiesDic },
             };
 
             return dict;
         }
 
-        internal static string GetUrl(this HttpRequest request)
-        {
-            if (request.Host.HasValue)
-            {
-                return $"{request.Scheme}://{request.Host.Value}{request.PathBase.ToUriComponent()}{request.Path.ToUriComponent()}";
-            }
-
-            // HTTP 1.0 requests are not required to provide a Host to be valid
-            // Since this is just for display, we can provide a string that is
-            // not an actual Uri with only the fields that are specified.
-            // request.GetDisplayUrl(), used above, will throw an exception
-            // if request.Host is null.
-            return $"{request.Scheme}://{HttpRequestExtensions.NoHostSpecified}{request.PathBase.ToUriComponent()}{request.Path.ToUriComponent()}";
-        }
+        internal static string GetUrl(this HttpRequest request, ImmutableTracerSettings settings = null) => HttpRequestUtils.GetUrl(request.Scheme, request.Host.Value, request.PathBase.ToUriComponent(), request.Path.ToUriComponent(), () => request.QueryString.Value, settings);
```

Comment:
nit: I would probably split this over several lines to make it easier to read.

Reply:
fixed in c5c4456c0eec985161bda650c99c0e468c5dc26b